lib/browser/api/menu.js

```diff
@@ -54,19 +55,38 @@ Menu.prototype.popup = function (window, x, y, positioningItem) {
     }
   }
 
+  // menu.popup({})
+  if (typeof window === 'object' && window.constructor !== BrowserWindow) {
+    opts = window
   // menu.popup(window, {})
-  if (x && typeof x === 'object') {
-    const opts = x
+  } else if (x && typeof x === 'object') {
+    opts = x
+  }
+
+  if (opts) {
     newX = opts.x
     newY = opts.y
     newPosition = opts.positioningItem
   }
 
   // set defaults
-  if (typeof x !== 'number') newX = -1
-  if (typeof y !== 'number') newY = -1
-  if (typeof positioningItem !== 'number') newPosition = -1
-  if (!window) newWindow = BrowserWindow.getFocusedWindow()
+  if (typeof newX !== 'number') newX = -1
+  if (typeof newY !== 'number') newY = -1
+  if (typeof newPosition !== 'number') newPosition = -1
+  if (!newWindow || (newWindow && newWindow.constructor !== BrowserWindow)) {
+    newWindow = BrowserWindow.getFocusedWindow()
+
+    // No window focused?
+    if (!newWindow) {
+      const browserWindows = BrowserWindow.getAllWindows()
+
+      if (browserWindows && browserWindows.length > 0) {
+        newWindow = browserWindows[0]
+      } else {
+        throw new Error(`Cannot open Menu without a BrowserWindow present`)
+      }
+    }
+  }
```

Comment:
This is more complicated before, and actually seems more complicated than the code was before #10888 as well.

So I'm trying to understand, is this a regression caused by #10888? If so, I think I'd prefer that we back that out + add your tests + *then* take another run at a refactor.

Alternately, was this broken before #10888 as well?

Reply:
This was broken _before_ 10888, too. We default to showing in the focused window, but a focused window might not be present. I took the liberty here of going with the first window if we can't find one, but we could easily continue to throw and take this portion out.

The important portion here is really that we don't override `newX` just because `x` isn't a number (and so on).

Reply:
I'm not disagreeing with the choice of taking the first window, was just was trying to understand if this was a 10888 regression because the pre-10888 code and the code here didn't do the same things.